lnwallet/errors.go

```diff
@@ -125,3 +125,49 @@ func ErrChanTooSmall(chanSize, minChanSize btcutil.Amount) ReservationError {
 			chanSize, minChanSize),
 	}
 }
+
+// ErrHtlcIndexAlreadyFailed is returned when the HTLC index has already been
+// failed, but has not been committed by our commitment state.
+type ErrHtlcIndexAlreadyFailed uint64
+
+// Error returns a message indicating the index that had already been failed.
+func (e ErrHtlcIndexAlreadyFailed) Error() string {
+	return fmt.Sprintf("HTLC with ID %d has already been failed", e)
+}
+
+// ErrHtlcIndexAlreadySettled is returned when the HTLC index has already been
+// settled, but has not been committed by our commitment state.
+type ErrHtlcIndexAlreadySettled uint64
```

Comment:
With this and the above, isn't it actually the case that the HTLC has already been settled or cancelled (_and_ it's update the commitment), but for some reason this was never ACK'd back?

Reply:
yes that is the case, i also added a concrete error type to handle the case where the htlc has been "modified", even though we don't currently handle it. right now we just ignore it bc we don't want to cleanup any responses until one gets committed